src/rules/matchers.ts

```diff
@@ -368,6 +374,68 @@ export class CookieMatcher extends Serializable implements RequestMatcher {
     }
 }
 
+export class CallbackMatcher extends Serializable implements RequestMatcher {
+    readonly type = "callback";
+  
+    constructor(
+        public callback: (request: OngoingRequest) => MaybePromise<boolean>
+        ) {
+        super();
+    }
+  
+    async matches(request: OngoingRequest) {
+        try {
+            return await this.callback(request);
+        } catch (error) {
+            return false;
+        }
+    }
+  
+    explain() {
+        return `matches using provided callback ${
```

Comment:
I think there shouldn't be a space here before the `$`, or we'll get a weird extra space in the explanations when there's no callback name.